modules/home-age.nix

```diff
@@ -0,0 +1,233 @@
+      '')
+      cfg.identityPaths;
+
+  cleanupAndLink = ''
+    _agenix_generation="$(basename "$(readlink "${cfg.secretsDir}")" || echo 0)"
+    (( ++_agenix_generation ))
+    echo "[agenix] symlinking new secrets to ${cfg.secretsDir} (generation $_agenix_generation)..."
+    ln -sfn "${cfg.secretsMountPoint}/$_agenix_generation" "${cfg.secretsDir}"
+
+    (( _agenix_generation > 1 )) && {
+    echo "[agenix] removing old secrets (generation $(( _agenix_generation - 1 )))..."
+    rm -rf "${cfg.secretsMountPoint}/$(( _agenix_generation - 1 ))"
+    }
+  '';
+
+  installSecrets = builtins.concatStringsSep "\n" (
+    [ "echo '[agenix] decrypting secrets...'" ]
+    ++ testIdentities
+    ++ (map installSecret (builtins.attrValues cfg.secrets))
+    ++ [ cleanupAndLink ]
+  );
+
+  secretType = types.submodule ({ config, name, ... }: {
+    options = {
+      name = mkOption {
+        type = types.str;
+        default = name;
+        description = ''
+          Name of the file used in ''${cfg.secretsDir}
+        '';
+      };
+      file = mkOption {
+        type = types.path;
+        description = ''
+          Age file the secret is loaded from.
+        '';
+      };
+      path = mkOption {
+        type = types.str;
+        default = "${cfg.secretsDir}/${config.name}";
+        description = ''
+          Path where the decrypted secret is installed.
+        '';
+      };
+      mode = mkOption {
+        type = types.str;
+        default = "0400";
+        description = ''
+          Permissions mode of the decrypted secret in a format understood by chmod.
+        '';
+      };
+      symlink = mkEnableOption "symlinking secrets to their destination" // { default = true; };
+    };
+  });
+
+  mountingScript =
+    let
+      app = pkgs.writeShellApplication {
+        name = "agenix-home-mount-secrets";
+        runtimeInputs = with pkgs; [ coreutils ];
+        text = ''
+          ${newGeneration}
+          ${installSecrets}
+          exit 0
+        '';
+      };
+    in
+    lib.getExe app;
+
+  userDirectory = dir:
+    let
+      inherit (pkgs.stdenv.hostPlatform) isDarwin;
+      baseDir =
+        if isDarwin
+        then "$(getconf DARWIN_USER_TEMP_DIR)"
+        else "$XDG_RUNTIME_DIR";
+    in
+    "${baseDir}/${dir}";
+
+  userDirectoryDescription = dir: ''
+    "$XDG_RUNTIME_DIR"/${dir} on linux or "$(getconf DARWIN_USER_TEMP_DIR)"/${dir} on darwin.
+  '';
+in
+{
+  imports = [
+    (mkRenamedOptionModule [ "age" "sshKeyPaths" ] [ "age" "identityPaths" ])
+  ];
+
+  options.age = {
+    package = mkPackageOption pkgs "rage" { };
+
+    secrets = mkOption {
+      type = types.attrsOf secretType;
+      default = { };
+      description = ''
+        Attrset of secrets.
+      '';
+    };
+
+    identityPaths = mkOption {
+      type = types.listOf types.path;
+      default =
+        if (config.services.openssh.enable or false)
```

Comment:
Is it going to be more compatible to use the user's SSH key instead of the host ssh key for decryption? Does this require the user to have root access?

Reply:
This is a copy-paste error, I intended to have an empty-list by default.